File: problems/ABC/ABC209/C.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    
    int N;
    long long MOD = pow(10, 9) + 7;
    cin >> N;
    vector<int> C(N);
    for (int i = 0; i < N; i++){
        cin >> C.at(i);
    }

    sort(C.begin(), C.end());

    long long ans, cnt;
    ans = 1;
    cnt = 0;

    for(int i = 0; i < N; i++){
        ans = ans * (C.at(i) - cnt);
        ans %= MOD;
        cnt++;
    }

    cout << ans << endl;
}

File: problems/ABC/ABC209/C.py
N = int(input())
C = list(map(int,input().split()))

C = sorted(C)

ans = 1
cnt = 0
for i in range(N):
    ans *= (C[i]-cnt)
    ans %= 10**9 + 7
    cnt += 1

print(ans)